actionpack/lib/action_controller/metal/strong_parameters.rb

```diff
@@ -531,6 +531,140 @@ def require(key)
 
     alias :required :require
 
+    # `mandate` is the preferred way to require and permit parameters.
+    # It is similar to calling `permit` and `require` in sequence, but it avoids some
+    # potential pitfalls, such as NoMethodErrors that can be triggered by external
+    # users passing unexpected scalar or non-scalar types.
```

Comment:
"scalar or non-scalar" doesn't read well IMO. Maybe:

```suggestion
    # users passing unexpected values.
```

```suggestion
    # users passing unexpected param types.
```